aws/sdk/integration-tests/s3/tests/presigning.rs

```diff
@@ -195,3 +195,32 @@ async fn test_presigned_head_object() {
         presigned.uri().to_string(),
     );
 }
+
+#[tokio::test]
+async fn customized_presigning() {
```

Comment:
It would be a good idea to add a test to Polly as well since it does request mutations in its presigning.